crates/core_simd/src/masks/bitmask.rs

```diff
@@ -115,22 +115,34 @@ where
         unsafe { Self(intrinsics::simd_bitmask(value), PhantomData) }
     }
 
-    #[cfg(feature = "generic_const_exprs")]
+    // Safety: N must be the exact number of bytes required to hold the bitmask for this mask
     #[inline]
     #[must_use = "method returns a new array and does not mutate the original value"]
-    pub fn to_bitmask(self) -> [u8; LaneCount::<LANES>::BITMASK_LEN] {
+    pub unsafe fn to_bitmask_array<const N: usize>(self) -> [u8; N] {
         // Safety: these are the same type and we are laundering the generic
         unsafe { core::mem::transmute_copy(&self.0) }
     }
 
-    #[cfg(feature = "generic_const_exprs")]
+    // Safety: N must be the exact number of bytes required to hold the bitmask for this mask
     #[inline]
     #[must_use = "method returns a new mask and does not mutate the original value"]
-    pub fn from_bitmask(bitmask: [u8; LaneCount::<LANES>::BITMASK_LEN]) -> Self {
+    pub unsafe fn from_bitmask_array<const N: usize>(bitmask: [u8; N]) -> Self {
         // Safety: these are the same type and we are laundering the generic
         Self(unsafe { core::mem::transmute_copy(&bitmask) }, PhantomData)
     }
 
+    // Safety: U must be the integer with the exact number of bits required to hold the bitmask for
+    #[inline]
+    pub unsafe fn to_bitmask_integer<U>(self) -> U {
+        unsafe { core::mem::transmute_copy(&self.0) }
+    }
+
+    // Safety: U must be the integer with the exact number of bits required to hold the bitmask for
+    #[inline]
+    pub unsafe fn from_bitmask_integer<U>(bitmask: U) -> Self {
+        unsafe { Self(core::mem::transmute_copy(&bitmask), PhantomData) }
+    }
```

Comment:
Publicly extending the API with a bunch of unsafe methods to implement what seemingly ought to be a simple operation is... less than ideal. I am also not certain that this is actually an inherently sound transmutation... it feels like it would break if `Simd<T, 17>` was legal (3 bytes), and I would prefer to not have to audit our internals later for such code if we relax the lane count restrictions, and while it is unsafe, an unsafe method that some masks could not use strikes me as an unnecessary footgun.

Reply:
This is not public.